src/clr-angular/data/datagrid/providers/selection.ts

```diff
@@ -39,44 +39,85 @@ export class Selection {
 
     this.subscriptions.push(
       this._items.allChanges.subscribe(updatedItems => {
-        if (!this._selectable) {
-          return;
-        }
-        let leftOver: any[] = this.current.slice();
-        let newSingle: any;
+        switch (this.selectionType) {
+          case SelectionType.None: {
+            break;
+          }
 
-        // Calculate the references for each item
-        const trackBy: TrackByFunction<any> = this._items.trackBy;
-        const matched = [];
-        updatedItems.forEach((item, index) => {
-          const ref = trackBy(index, item);
-          // Look in current selected refs array if item is selected, and update actual value
-          if (this.selectedSingle === ref) {
-            newSingle = item;
-          } else if (this.selected.length) {
-            const selectedIndex = this.selected.indexOf(ref);
-            if (selectedIndex > -1) {
-              matched.push(selectedIndex);
-              leftOver[selectedIndex] = item;
+          case SelectionType.Single: {
+            let newSingle: any;
+            const trackBy: TrackByFunction<any> = this._items.trackBy;
+            let selectionUpdated: boolean = false;
+
+            updatedItems.forEach((item, index) => {
+              const ref = trackBy(index, item);
+              // If one of the updated items is the previously selectedSingle, set it as the new one
+              if (this.prevSingleSelectionRef === ref) {
+                newSingle = item;
+                selectionUpdated = true;
```

Comment:
Ok thinking about the cases here, and I _think_ this is what will happen. In all cases, if you don't select an item and reload the list, then it will not emit (as expected). However, if you select and reload the list, I believe the behaviors are as follows.

1. ngFor, no trackBy (or trackBy index). If you have already selected an item (which is by index), any reloading of the dataset would essentially select the new item at that index regardless if it was the original item or not. Do we address this by always clearing the selection in this case on reload, or is this not a supported use case? 
2. ngFor, trackBy. If you already have selected an item (which is track by the item), and reload the dataset, if the reference exists it will set and emit it again. If the reference doesn't exist, it will do nothing. Outside of the extra emit, this looks correct.
3. clrDgItems, no trackBy (or trackBy index). If you select an item (tracked by index), and you reload, we run into the same problem as 1 where the index will remain stored and it will select the new item at that index and emit it. If the reference doesn't exist (as in the index is larger than the length of the new list), no refs would match and it would be set to `undefined` and emitted. 
4. clrDgItems, trackBy. If you select an item (track by item) and reload, if the reference exists in the new list the ref will match and the selection will be emitted again. If the reference doesn't exist, it is cleared and set to `undefined`. Outside of the extra emit, this looks correct.

So the problems I see:

1. If a new set of items is loaded and the trackBy (index or item) matches an item it will be emitted again (since emit happens in the `currentSingle` setter). Do we have an expectation that the event will _only_ fire when a change actually happens, or is this an acceptable case? I think we can live with it, as a limitation where apps will have to handle possible multiple emits of the same selected value (or it should be an idempotent event handler). 
2. If we have a computed reference that is an index, then reloading the list will automatically set `currentSingle` to the new item at that index (regardless if it is the same item or not). This is a new issue not originally outlined. If you aren't using the smart iterator, then perhaps we should always clear the `currentSingle` value, and if you don't want that behavior then use a trackBy item calculation?

Reply:
Talked with Jeeyun more, while both are quirky cases, they are generally no the ideal case and are best handled with having better documentation about how to use trackBy and selection modes.